demos/esc/Kconfig

```diff
@@ -0,0 +1,25 @@
+# ESC (Electronic Speed Controller) demo configuration
+
+mainmenu "Eurus Flight Controller - ESC demo"
+
+
+# ESC Protocol configuration
+choice
+    prompt "ESC Protocol Selection"
+    default ESC_PWM
+
+config ESC_PWM
+    bool "ESC PWM protocol"
+
+config ESC_ONESHOT_125
+    bool "ESC Oneshot125 protocol"
+
+config ESC_ONESHOT_42
+    bool "ESC Oneshot42 protocol"
+
+config ESC_MULTISHOT
+    bool "ESC Multishot protocol"
+
+endchoice
+
+source "../efc/libs/motor_control/esc/Kconfig"
```

Comment:
If the intent is for us to use this method of sourcing Kconfig files for the time being, this should also be present in the main firmware.

Long-term I guess we want to use a more reliable method (perhaps Zephyr modules), since it is easy to forget to do this and we shouldn't need to both add a library in CMake and source Kconfig as separate steps.